Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGppProtobufExtensions.cs

```diff
@@ -0,0 +1,610 @@
+                    {
+                        Value = observedAccessedFileName.Value
+                    });
+                }
+
+                foreach (var priorStrongFingerprint in strongFingerprintComputation.PriorStrongFingerprints)
+                {
+                    processStrongFingerprintComputationData.PriorStrongFingerprints.Add(new StrongContentFingerPrint_XLGpp()
+                    {
+                        Hash = priorStrongFingerprint.Hash.ToFingerprint_XLGpp()
+                    });
+                }
+
+                foreach (var observedInput in strongFingerprintComputation.ObservedInputs)
+                {
+                    processStrongFingerprintComputationData.ObservedInputs.Add(new ObservedInput_XLGpp()
+                    {
+                        Type = (ObservedInputType_XLGpp)observedInput.Type,
+                        Hash = new ContentHash_XLGpp()
+                        {
+                            Value = observedInput.Hash.ToString()
+                        },
+                        PathEntry = observedInput.PathEntry.ToObservedPathEntry_XLGpp(pathTable),
+                        Path = observedInput.Path.ToAbsolutePath_XLGpp(pathTable),
+                        IsSearchPath = observedInput.IsSearchPath,
+                        IsDirectoryPath = observedInput.IsDirectoryPath,
+                        DirectoryEnumeration = observedInput.DirectoryEnumeration
+                    });
+                }
+
+                processFingerprintComputationEvent.StrongFingerprintComputations.Add(processStrongFingerprintComputationData);
+            }
+
+            return processFingerprintComputationEvent;
+        }
+
+        /// <nodoc />
+        public static ExtraEventDataReported_XLGpp ToExtraEventDataReported_XLGpp(this ExtraEventData data, uint workerID)
+        {
+            var Uuid = Guid.NewGuid().ToString();
+
+            return new ExtraEventDataReported_XLGpp
+            {
+                UUID = Uuid,
+                WorkerID = workerID,
+                DisableDetours = data.DisableDetours,
+                IgnoreReparsePoints = data.IgnoreReparsePoints,
+                IgnorePreloadedDlls = data.IgnorePreloadedDlls,
+                ExistingDirectoryProbesAsEnumerations = data.ExistingDirectoryProbesAsEnumerations,
+                NtFileCreateMonitored = data.NtFileCreateMonitored,
+                ZwFileCreateOpenMonitored = data.ZwFileCreateOpenMonitored,
+                IgnoreZwRenameFileInformation = data.IgnoreZwRenameFileInformation,
+                IgnoreZwOtherFileInformation = data.IgnoreZwOtherFileInformation,
+                IgnoreNonCreateFileReparsePoints = data.IgnoreNonCreateFileReparsePoints,
+                IgnoreSetFileInformationByHandle = data.IgnoreSetFileInformationByHandle,
+                IgnoreGetFinalPathNameByHandle = data.IgnoreGetFinalPathNameByHandle,
+                FingerprintVersion = (int)data.FingerprintVersion,

```

Comment:
>(int) [](start = 37, length = 5)

why is casting needed?

Reply:
Since we cannot have an enum starting at non-zero value. Thus the alternative is to just store as an int.

---
In reply to: [308904056](https://github.com/microsoft/BuildXL/pull/671#discussion_r308904056) [](ancestors = 308904056)